test/tests/userobjects/shape_element_user_object/tests

```diff
@@ -79,4 +79,27 @@
       detail = 'for a full physics test.'
     []
   []
+  [diagonal]
+    requirement = 'The system shall error if a user attempts to use nonlocal residual objects without appropriate coupling specified in a preconditioning object. These residual objects include'
+    [kernels]
+      type = RunException
+      input = 'jacobian.i'
+      expect_err = 'Nonlocal kernels only supported for non-diagonal coupling'
+      detail = 'kernels'
+      cli_args = "Preconditioning/active='' Executioner/solve_type=PJFNK"
+      allow_warnings = true
+      recover = false
+      max_threads = 1
+    []
+    [bcs]
+      type = RunException
+      input = 'shape_side_uo_jac_test.i'
+      expect_err = 'Nonlocal boundary conditions only supported for non-diagonal coupling'
+      detail = 'boundary conditions'
+      cli_args = "Preconditioning/active='' Executioner/solve_type=PJFNK"
+      allow_warnings = true
+      recover = false
+      max_threads = 1
+    []
+  []
```

Comment:
This looks good to me. Should we add another test with `automatic_scaling = true`, making sure that we don't hit the error by accident?

Reply:
Oh yea ... I think that we will indeed hit that error 😆 I believe I can find a good work-around for that